packages/application-components/src/components/modal-pages/internals/modal-page.styles.ts

```diff
@@ -1,26 +1,50 @@
 import { css, type SerializedStyles } from '@emotion/react';
 import { customProperties } from '@commercetools-uikit/design-system';
-
-export const TRANSITION_DURATION = 200;
+import type { TModalPageSize } from './modal-page';
 
 type StyleProps = {
   zIndex?: number;
-  size?: 'small' | 'large';
+  size: TModalPageSize;
+};
+
+export const stylesBySize = {
+  '10': {
+    width: '600px',
+    transitionTime: 200,
+  },
+  '20': {
+    width: '900px',
+    transitionTime: 200,
+  },
+  '30': {
+    width: '1200px',
+    transitionTime: 200,
+  },
+  scale: {
+    width: '100%',
+    transitionTime: 300,
+  },
 };
 
 export const getContainerStyles = (_props: StyleProps): SerializedStyles => css`
   position: absolute;
   top: 0;
   right: 0;
   height: 100%;
-  width: ${_props.size === 'small' ? '600px !important' : '100%'};
+  width: ${_props.size !== 'scale'
```

Comment:
Nit: I know it was already like this but could you rename it to `props` instead of `_props`? In TS an underscore as a prefix is used to silent the compiler that a variable is unused.

Thanks